src/lib/WalletInfoCollector.ts

```diff
@@ -92,14 +92,16 @@ export default class WalletInfoCollector {
             derivedAccountsPromise = WalletInfoCollector._deriveAccounts(startIndex,
                 ACCOUNT_MAX_ALLOWED_ADDRESS_GAP, walletType, walletId);
 
-            // find accounts with a balance > 0
-            // TODO should use transaction receipts
+            // find accounts with a balance > 0 and/or transaction receipts
             foundAccounts = [];
             const balances = await WalletInfoCollector._getBalances(derivedAccounts!);
             for (const account of derivedAccounts!) {
                 const balance = balances.get(account.address);
-                if (balance === undefined || balance === 0) continue;
-                foundAccounts.push(account);
+                const hasBalance = balance !== undefined && balance !== 0;
+                const receipts = await NetworkClient.Instance.requestTransactionReceipts(account.address);
+                if (hasBalance || receipts.length > 0) {
```

Comment:
Please only get receipts when `hasBalance` is false, as the receipt getting creates unnecessary network congestion and delay otherwise.

Another problem we will encounter is rate-limiting for receipts getting. `FullConsensusAgent` has a rate-limit of 30/min for transaction-receipts requests: https://github.com/nimiq-network/core/blob/master/src/main/generic/consensus/full/FullConsensusAgent.js#L469, https://github.com/nimiq-network/core/blob/master/src/main/generic/consensus/full/FullConsensusAgent.js#L591.

Thus the question is if we can switch the peer that we request from, if we have another full peer available.

This brings me to the third problem: Receipts getting only works, when we are actually connected to a full node (which is not always the case, as found in the Safe sometimes), so this needs error handling.

Reply:
Puh. Sounds like stuff that should be handled elsewhere...